src/Compilers/CSharp/Portable/Symbols/MethodSymbolExtensions.cs

```diff
@@ -190,6 +191,48 @@ public static bool IsAsyncReturningIAsyncEnumerator(this MethodSymbol method, CS
                 && method.ReturnType.IsIAsyncEnumeratorType(compilation);
         }
 
+#nullable enable
+        public static bool IsAsyncReturningTaskViaOverride(this MethodSymbol method, [NotNullWhen(true)] out object? builderArgument)
```

Comment:
>IsAsyncReturningTaskViaOverride [](http://example.com/codeflow?start=27&length=31)

Are `IsAsyncReturningGenericTask()` and `IsAsyncReturningTaskViaOverride()` ever called separately? If not, consider combining. Same question for generic task pair.